Lib/test/support/_hypothesis_stubs/__init__.py

```diff
@@ -0,0 +1,71 @@
+import functools
+import unittest
+
+__all__ = [
+    "given",
+    "example",
+    "assume",
+    "reject",
+    "register_random",
+    "strategies",
+    "HealthCheck",
+    "settings",
+    "Verbosity",
+]
+
+from . import strategies
+from ._settings import HealthCheck, Verbosity, settings
+from .control import assume, reject
+
+
+def given(*_args, **_kwargs):
+    def decorator(f):
+        if getattr(f, "_given", False):
+            # given can be used more than once, but for our purposes there's
+            # no need to wrap the function multiple times.
+            return f
+
+        if examples := getattr(f, "_examples", []):
+
+            @functools.wraps(f)
+            def test_function(self):
+                for example_args, example_kwargs in examples:
+                    with self.subTest(*example_args, **example_kwargs):
+                        f(self, *example_args, **example_kwargs)
+
+        else:
+            test_function = unittest.skip(
```

Comment:
It would be nice to add a comment here too that `@example` will re-wrap the original function if applied after this `@given` decorator.

Reply:
Done.